src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,25 +8,47 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+//    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+//    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    //public static final String ADDRESS_VALIDATION_REGEX = ".+";
+
+    //modify
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses should be in BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "^\\d{1,4}\\,[A-Za-z0-9.]*\\,\\#\\d{1,2}\\-\\d{1,3}\\,\\d{6}$";    //modify
 
     public final String value;
     private boolean isPrivate;
 
+    //modification: block, street, unit, postal code
+    int block;
```

Comment:
Should class level abstraction be in `String` and `int`? 

Recall OOP concepts and example on abstraction during tutorial. 
(A Bicycle object consist of multiple lower-level objects such as 'wheels', 'frame' and 'gears')